move/its/sources/discovery.move

```diff
@@ -66,9 +68,16 @@ module its::discovery {
         ));
     }
 
-    public fun get_call_info(self: &ITS, payload: vector<u8>): Transaction {
+    public fun get_call_info(self: &ITS, mut payload: vector<u8>): Transaction {
         let mut reader = abi::new_reader(payload);
-        let message_type = reader.read_u256();
+        let mut message_type = reader.read_u256();
+
+        if (message_type == MESSAGE_TYPE_RECEIVE_FROM_HUB) {
+            reader.skip_slot();
+            payload = reader.read_bytes();
+            reader = abi::new_reader(payload);
+            message_type = reader.read_u256();
+        };
 
         if (message_type == MESSAGE_TYPE_INTERCHAIN_TRANSFER) {
             get_interchain_transfer_tx(self, &mut reader)
```

Comment:
```suggestion
            return get_interchain_transfer_tx(self, &mut reader)
```
return early, so else nesting isn't needed

Reply:
Can you please explain? I thought in rust if you had an if statement at the end of a function without a semicolon, whatever the first successful branch returns is returned as well, meaning changing to what you suggest makes zero difference. Where am I wrong?

Reply:
You cannot have the first one, since the last branch does not return the right type